rust-runtime/aws-smithy-client/src/lib.rs

```diff
@@ -78,7 +78,13 @@ pub mod conns {
 
     #[cfg(feature = "native-tls")]
     pub fn native_tls() -> NativeTls {
-        hyper_tls::HttpsConnector::new()
+        let mut tls = hyper_tls::native_tls::TlsConnector::builder();
+        let tls = tls
+            .min_protocol_version(Some(hyper_tls::native_tls::Protocol::Tlsv12))
+            .build()
+            .unwrap_or_else(|e| panic!("Error while creating TLS connector: {}", e));
+        let mut http = hyper::client::HttpConnector::new();
+        hyper_tls::HttpsConnector::from((http, tls.into()))
```

Comment:
This will still allow `http` connections when supplying an `http` URL. 
Maybe a comment should be added to the `native_tls()` function to make users aware of that?

Reply:
Added documentation to both `native_tls` and `https` in https://github.com/awslabs/smithy-rs/pull/2312/commits/97b3d33fcd0d8b2041531767f073f38121cf685b - do you think it's clear enough?

Reply:
Yes it's very clear, thank you! :)